### DISCUSSION TITLE
PixelExperience a12 Random Errors

### DISCUSSION THREAD

**Discussion by ImSurajx** [ROM-builders/temporary#14517] [Show and tell]
error: prebuilts/clang/host/linux-x86/Android.bp:531:1: module "libclang_rt.ubsan_minimal-x86_64" variant "linux_glibc_x86_64_static": module source path "prebuilts/clang/host/linux-x86/clang-r416183b1/lib64/clang/12.0.7/lib/linux/libclang_rt.ubsan_minimal-x86_64.a" does not exist
error: prebuilts/clang/host/linux-x86/Android.bp:449:1: module "libclang_rt.ubsan_standalone-x86_64.static" variant "linux_glibc_x86_64_static": module source path "prebuilts/clang/host/linux-x86/clang-r416183b1/lib64/clang/12.0.7/lib/linux/libclang_rt.ubsan_standalone-x86_64.a" does not exist

Cirrus Link :
https://cirrus-ci.com/build/5802361231245312
https://cirrus-ci.com/build/6371043219603456
https://cirrus-ci.com/build/6455000703434752

**Comment by Apon77** [comment]
What can be done to fix?

**Reply by ImSurajx** [reply]
Resync Source. My Mean Clean Build Required

**Reply by Alone0316** [reply]
> What can be done to fix?

Do clean sync and ask everyone to stop cloning that clang, those who ever is cloning that clang his builds get succeeds but everyone else who makes that ROM after him will get error and only solution is clean sync or either try implementing some script which checks all the stuff which was cloned by person X and remove it when Person Y is building

**Reply by ImSurajx** [reply]
YuPs But Who One Gonna Make It, If It Is Possible By Manifest Then It's is Good For Us..

**Reply by Alone0316** [reply]
> YuPs But Who One Gonna Make It, If It Is Possible By Manifest Then It's is Good For Us..

We can have one more step in ci before syncing which can be named as pre sync or cleanup in which it should grep stuff cloned by person X and remove them and then start sync for person Y

**Reply by ImSurajx** [reply]
I Guess It's Part Of Sync, You Can Do This Kind Of Stuff By Sync. like Remove Project & Adding Project

**Reply by Alone0316** [reply]
> I Guess It's Part Of Sync, You Can Do This Kind Of Stuff By Sync. like Remove Project & Adding Project

while syncing it clones the local manifest first and then sync the stuff so if we implement it in sync then person X's manifest will be replaced by person Y's manifest so we won't have access to person X's manifest who is cloning it

**Reply by ImSurajx** [reply]
Like Sync Proccess Remove Old Device Tree Before Syncing New Device Tree?

**Reply by Alone0316** [reply]
> Like Sync Proccess Remove Old Device Tree Before Syncing New Device Tree?

Yes but it isn't able to remove that clang and as i said the script will check person X's manifest and remove the stuff and then sync the source so if we implement it in sync then as the local manifest gets cloned first and then it syncs then we will loose person X's manifest as it will get replaced by person Y's manifest so it won't be able to check the stuff which was synced by person X
